contrib/go/src/python/pants/contrib/go/rules/targets.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    Sources,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+
+
+class GoSources(Sources):
+    # NB: We glob on `*` due to the way resources and .c companion files are handled.
+    default = ("*", "!BUILD", "!BUILD.*")
+
+
+class BuildFlags(StringField):
+    """Build flags to pass to the Go compiler."""
+
+    alias = "build_flags"
+
+
+class GoBinary(Target):
+    """A Go main package."""
+
+    alias = "go_binary"
+    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, GoSources, BuildFlags)
+
+
+class GoLibrary(Target):
+    """A Go package."""
+
+    alias = "go_library"
+    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, GoSources)
+
+
+class GoProtobufSources(Sources):
+    default = ("*.proto",)
+
+
+class ProtocPlugins(StringSequenceField):
+    """Protoc plugins to use when generating code from this target."""
+
+    alias = "protoc_plugins"
+
+
+class GoProtobufLibrary(Target):
+    """A Go library generated from Protobuf IDL files."""
+
+    alias = "go_protobuf_library"
+    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, GoProtobufSources, ProtocPlugins)
+
+
+class GoThriftLibrary(Target):
+    """A Go library generated from Thrift IDL files."""
+
+    alias = "go_thrift_library"
+    core_fields = (*COMMON_TARGET_FIELDS, Dependencies, Sources)
+
+
+class Package(StringField):
```

Comment:
nit: for this and revision below, they seem pretty go-centric contepts (more for this than revsion, but still). To avoid confusion at usage sites, could you name this `GoPackage`?

Reply:
Good call! Not at all a nit and this is one of my favorite aspects of the Target API, that you can call the Python symbol whatever you want and still use a certain `alias`.